Lib/plistlib.py

```diff
@@ -217,14 +218,25 @@ def get_data(self):
 
     def begin_dict(self, attrs):
         d = self._dict_type()
+        self.previous_dict_key = self.current_key
         self.add_object(d)
         self.stack.append(d)
 
     def end_dict(self):
         if self.current_key:
             raise ValueError("missing value for key '%s' at line %d" %
                              (self.current_key,self.parser.CurrentLineNumber))
-        self.stack.pop()
+        d = self.stack.pop()
+        # Unmarshal CF$UID to UID and replace the dict node
+        if len(d) == 1 and "CF$UID" in d and isinstance(d["CF$UID"], int):
+            uid = UID(d["CF$UID"])
+            if self.previous_dict_key:
+                self.stack[-1][self.previous_dict_key] = uid
+            elif not self.stack:
+                self.root = uid
+            else:
+                assert isinstance(self.stack[-1], type([]))
```

Comment:
I believe `type([])` should be replaced with `list`.

In any case, this `assert` seems a bit suspicious to me.